sensorprocess/sensorprocess.go

```diff
@@ -47,24 +48,25 @@ func Start(
 func addSensorReading(
 	ctx context.Context,
 	config Config,
+	isOnline bool,
 ) bool {
 	tsr, err := config.Lidar.TimedSensorReading(ctx)
 	if err != nil {
 		config.Logger.Warn(err)
 		return strings.Contains(err.Error(), replaypcd.ErrEndOfDataset.Error())
 	}
-	if tsr.Replay {
-		addSensorReadingFromReplaySensor(ctx, tsr.Reading, tsr.ReadingTime, config)
-	} else {
-		timeToSleep := addSensorReadingFromLiveReadings(ctx, tsr.Reading, tsr.ReadingTime, config)
+	if isOnline {
+		timeToSleep := addSensorReadingOnline(ctx, tsr.Reading, tsr.ReadingTime, config)
 		time.Sleep(time.Duration(timeToSleep) * time.Millisecond)
+	} else {
+		addSensorReadingOffline(ctx, tsr.Reading, tsr.ReadingTime, config)
 	}
 	return false
 }
 
-// addSensorReadingFromReplaySensor adds a reading from a replay sensor to the cartofacade
+// addSensorReadingOffline adds a reading to the cartofacade
 // retries on error.
-func addSensorReadingFromReplaySensor(ctx context.Context, reading []byte, readingTime time.Time, config Config) {
+func addSensorReadingOffline(ctx context.Context, reading []byte, readingTime time.Time, config Config) {
```

Comment:
What do you think of calling this tryAddSensorReadingUntilSuccess?

Also, it might be possible to write this in terms of tryAddSensorReading (but not a big deal if not or if it ends up looking less clear that way)

Reply:
updating!

Reply:
I think writing ` tryAddSensorReadingUntilSuccess` in terms of ` tryAddSensorReading` ends up being a little less clear because really they are both wrappers around `CartoFacade.AddSensorReading`

Reply:
ok, sounds good!